### PR TITLE
Document Signal events

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## What is the purpose of the change

This PR documents the top-level signal start event and adds a short description about the Signals concept.

closes https://github.com/camunda/zeebe/issues/11198

## When should this change go live?

Release 8.2

## PR Checklist

- [ ] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/components/concepts/signals.md
Comment (korthout) [inline docs/components/concepts/signals.md:8]: 🔧 Let's start by introducing what signals are: e.g. Signals are similar to Messages. But they are also different because they trigger all matching signal events, where messages are correlated to a specific process instance.

🔧 Let's also avoid referring to Camunda Platform 8 instead of Zeebe, because Signals also exist in Web and Desktop Modeler, Operate, and Optimize.

**Review by korthout — CHANGES_REQUESTED** [review]
Great work @remcowesterhoud ✍️ 

There are a few things that should be adjusted. I also wrote a bunch of suggestions. Please have a look.

File: sidebars.js
Comment (remcowesterhoud) [reply sidebars.js:76]: I wouldn't want to merge it to be honest. I think there are too much differences and the page would become a mess explaining what is relevant for message and what is relevant for signals. I can definitely move it closer though!

File: docs/components/concepts/signals.md
Comment (remcowesterhoud) [reply docs/components/concepts/signals.md:14]: > While this is correct, we don't yet support this behavior. Currently, only the top-level signal start event is supported, which behaves differently. A signal subscription is opened when a process is deployed with a top-level signal start event.

I'm not sure I agree that this is a problem. We document the concept of Signals on this page. We shouldn't care about the elements and if they are support or not. That's why we have these different pages.

> We should also document what happens when a new process version is deployed (subscription is closed).

This is specific to the top-level start event. That's why it's documented on that page, instead of on this one. As I said above I think the concept of signals is separate from the events and what you mention here is in my opinion an event detail, not a concept detail.

We also don't document these things in the concept page of messages.

**Comment by korthout** [comment]
@remcowesterhoud You probably have to rebase on top of main and copy all the changes to the `version-8.2` versioned docs 😢

**Comment by remcowesterhoud** [comment]
> @remcowesterhoud You probably have to rebase on top of main and copy all the changes to the version-8.2 versioned docs 😢

Not sure where you're looking, but this version doesn't exist yet. I'd assume this won't be created until the 8.2 release.

**Comment by korthout** [comment]
> Not sure where you're looking, but this version doesn't exist yet. I'd assume this won't be created until the 8.2 release.

Sorry for the confusion. I had seen a comment about this, but the 8.2 docs haven't been cut yet.

**Review by christinaausley — APPROVED** [review]
Technical review in latest commit 👍

**Comment by akeller** [comment]
@remcowesterhoud @korthout, you are good to merge, but based on the conversation in the PR I wasn't sure if this was ready 😅

File: docs/components/concepts/signals.md
Comment (korthout) [inline docs/components/concepts/signals.md]: ❓ Should we add a remark that a signal subscription is also opened for a deployed process with signal start event? Just for completeness.

File: docs/components/modeler/bpmn/signal-events/signal-events.md
Comment (korthout) [inline docs/components/modeler/bpmn/signal-events/signal-events.md:27]: ❌ We need to expand this by explaining that the merging behavior can be customized using output mappings.

> When broadcasting a signal you can pass along variables. By default, all signal variables are merged into the process instance. This behavior can be customized by defining an output mapping at the signal catch event.

**Review by korthout — APPROVED** [review]
Thanks @remcowesterhoud and @christinaausley 

Please have a look at my comments before merging. I have one last change request and a question, but I see no reason why this could not be merged.

LGTM 👍

**Comment by remcowesterhoud** [comment]
@akeller Could I ask you for an approval here? All the feedback has been discussed and processed, but the approval I had became stale after a new commit 😞 

Not a big fan of commits removing approvals 😛

**Review by akeller — APPROVED** [review]
> Not a big fan of commits removing approvals 😛

Same. Need to see if we can get a better balance after this minor release.

**Pull request merged**
